src/frontend/staff-portal/src/app/components/jj-dispute-info/jj-dispute/jj-dispute.component.ts

```diff
@@ -233,10 +233,10 @@ export class JJDisputeComponent implements OnInit {
     element?.nativeElement.scrollIntoView({ behavior: 'smooth' });
   }
 
-  onConfirm(): void {
+  onConfirm(): void {    
     // TCVP-3082: Set dueDate values to current date before opening the dialog
     this.lastUpdatedJJDispute.jjDisputedCounts.forEach(count => {
-      if (!count.revisedDueDate) {
+      if (!count.revisedDueDate && count.jjDisputedCountRoP.finding === JJDisputedCountRoPFinding.Guilty) {
         count.revisedDueDate = new Date().toISOString();

```

Comment:
Perhaps grab the date outside of the loop to ensure each count would have the exact same revised date. The seconds could rollover in the middle of the loop.